# A1 · Mi base de datos

**Nombre:** Estefanía Rubí Jarquín Toledo

**Carrera:** Negocios Internacionalees

---

**Antes de empezar:** Entorno de ejecución → Cambiar tipo de entorno de ejecución
→ **R**, y después Archivo → Guardar una copia en Drive.

Este cuaderno es el inicio de tu portafolio. La base que elijas aquí es la que vas
a analizar el resto del semestre, así que vale la pena elegirla con cuidado.

Las celdas de texto con *Escribe aquí* son las que contestas. Borra esa indicación
y escribe tu respuesta en su lugar.

## 0 · Paquetes

La primera celda revisa qué paquetes faltan en la máquina de Colab y solo instala
esos. Se corre siempre al abrir el cuaderno.

In [3]:
paquetes <- c("tidyverse", "readxl", "haven")

faltantes <- paquetes[!paquetes %in% rownames(installed.packages())]
if (length(faltantes) > 0) install.packages(faltantes)

library(tidyverse)
library(readxl)   # archivos de Excel
library(haven)    # archivos de SPSS y Stata

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.2.1     ✔ readr     2.2.0
✔ forcats   1.0.1     ✔ stringr   1.6.0
✔ ggplot2   4.0.3     ✔ tibble    3.3.1
✔ lubridate 1.9.5     ✔ tidyr     1.3.2
✔ purrr     1.2.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


## 1 · Tu base

Elige una fuente del
[catálogo de datos](https://cjjmdata.github.io/analisis_datos_i/datos/catalogo.html).
Puede ser cualquiera, no solo las sugeridas para tu carrera.

Antes de elegir, revisa en las instrucciones de la actividad en Brightspace **cómo
se descarga** cada fuente. Algunas llegan en un solo archivo y otras en un
comprimido con varias tablas.

| | |
|---|---|
| **Fuente** (como aparece en el catálogo) | Banco Mundial, PIB |
| **Archivo** (nombre exacto, con extensión) | API_NY.GDP.MKTP.CD_DS2_en_excel_v2_375029.xls |
| **Dirección de descarga** | https://api.worldbank.org/v2/en/indicator/NY.GDP.MKTP.CD?downloadformat=excel |
| **Fecha en que lo descargaste** | 15 de Septiembre de 2026 |

Si la fuente no tiene una dirección directa al archivo, escribe la dirección de
la página y los pasos que seguiste para descargarlo.

## 2 · Tu pregunta

### Por qué elegiste esta base

Elegí esta base de datos porque sí analizo el Producto Interno Bruto (PIB) podré entender mejor la economía del país y como ha ido evolucionando al paso del tiempo. Creo yo que como estudiante de Negocios Internacionales, es fundamental saber y comprender el contexto de la economía y saber que hace que varie tanto y que define que cosa

### La pregunta que quieres responder

¿Cómo  ha ido evoluacionado el Producto Interno Bruto en México?

### Las variables que necesitarías

- Nombre del país
- Código ISO del país
- Columnas de los años donde estan nos valores del PIB



---

Una pregunta que este curso puede responder cumple tres condiciones:

1. **Es descriptiva**: pregunta cuánto, cómo se reparte o cómo se compara entre
   grupos. No pregunta qué causa qué.
2. **Se responde con las variables de tu base**, no con datos que no tienes.
3. **Habla de lo que tu base cubre**: sus unidades, su periodo, su geografía.

Con la encuesta del grupo como ejemplo:

| Pregunta | ¿Sirve? | Por qué |
|---|:--:|---|
| ¿Qué carrera concentra más estudiantes en el grupo? | sí | descriptiva; usa `carrera` |
| ¿El tiempo de traslado es parecido entre carreras? | sí | compara grupos; usa `traslado` y `carrera` |
| ¿Vivir lejos hace que un alumno saque peores calificaciones? | no | pregunta por una causa, y la base no tiene calificaciones |
| ¿Cuánto mide en promedio un estudiante de la universidad? | no | la base es un grupo, no la universidad |

## 3 · Leer la base

La función depende de la extensión del archivo. La tabla completa y los detalles
de cada formato están en la
[referencia de R](https://cjjmdata.github.io/analisis_datos_i/curso/referencia-r.html#abrir-un-archivo-de-datos).

Guarda la base en un objeto llamado **`serie`**. Es el nombre que usan todas las
sesiones del curso para tu base.

Tres situaciones frecuentes:

**El archivo está en una dirección de internet y es `.csv` o `.csv.gz`**

```r
serie <- read_csv("https://direccion/del/archivo.csv.gz")
```

**El archivo está en internet y es Excel, SPSS o un `.zip`**: primero se descarga
a la máquina de Colab.

```r
download.file("https://direccion/del/archivo.zip", "archivo.zip", mode = "wb")
unzip("archivo.zip")
list.files(recursive = TRUE)      # qué archivos salieron del zip, también dentro de carpetas
serie <- read_csv("nombre_del_archivo.csv")
```

**Descargaste el archivo a tu computadora**: súbelo con el icono de carpeta de la
barra izquierda, y léelo por su nombre.

```r
serie <- read_excel("archivo.xlsx")
```

Escribe tu lectura en la celda de abajo.

In [5]:
# Lectura de la base
download.file("https://api.worldbank.org/v2/en/indicator/NY.GDP.MKTP.CD?downloadformat=excel", "pib_mexico.xls", mode = "wb")
serie <- read_excel("pib_mexico.xls", skip = 3)

Si R imprime *"One or more parsing issues"*, la tabla sí se leyó, pero en alguna
columna R adivinó mal el tipo y dejó valores como `NA`. Vuelve a leer agregando
`guess_max = Inf`:

```r
serie <- read_csv("nombre_del_archivo.csv", guess_max = Inf)
```

## 4 · La estructura de tu base

In [6]:
glimpse(serie)

Rows: 265
Columns: 70
$ `Country Name`   <chr> "Aruba", "Africa Eastern and Southern", "Afghanistan"…
$ `Country Code`   <chr> "ABW", "AFE", "AFG", "AFW", "AGO", "ALB", "AND", "ARB…
$ `Indicator Name` <chr> "GDP (current US$)", "GDP (current US$)", "GDP (curre…
$ `Indicator Code` <chr> "NY.GDP.MKTP.CD", "NY.GDP.MKTP.CD", "NY.GDP.MKTP.CD",…
$ `1960`           <dbl> NA, 24205729121, NA, 11904805742, NA, NA, NA, NA, NA,…
$ `1961`           <dbl> NA, 24958927694, NA, 12707719291, NA, NA, NA, 1999843…
$ `1962`           <dbl> NA, 27073279053, NA, 13630588814, NA, NA, NA, 2032904…
$ `1963`           <dbl> NA, 31769188044, NA, 14468913698, NA, NA, NA, 2236447…
$ `1964`           <dbl> NA, 30279603755, NA, 15803564796, NA, NA, NA, 2481330…
$ `1965`           <dbl> NA, 33806233153, NA, 16920876930, NA, NA, NA, 2677110…
$ `1966`           <dbl> NA, 36927311239, NA, 18033947667, NA, NA, NA, 2835104…
$ `1967`           <dbl> NA, 38444227753, NA, 16493541544, NA, NA, NA, 3060086…
$ `1968`          

### Registros y variables

- Registros (filas): 265
- Variables (columnas): 70

### Unidad de observación

Cada renglón representa un país o una región económica del mundo (ya sea México, Aruba, etc.).

### Variables

Elige de **3 a 5 variables**, incluidas las que nombraste en tu pregunta.

El **significado** sale del diccionario de datos, del descriptor de archivos o de
la página de la fuente, no del nombre de la columna. Escribe de dónde lo sacaste.

El **tipo en R** es el que aparece entre `< >` en la salida de `glimpse()`.
El **tipo estadístico** es el que vimos en la sesión 4: categórica nominal,
categórica ordinal, numérica discreta o numérica continua.

| Variable | Qué significa | Tipo en R | Tipo estadístico |
|---|---|---|---|
| Country name | Nombre del país o región | <chr> | Categórica nominal |
| Country Code| Código ISO del identificacion del país | <chr> | Categórica nominal |
| 2024 | Valor del Producto Interno Bruto en dólares estadounidenses | <dbl> | Númerica continua |

*Fuente del significado de las variables: Metadatos oficiales y documentación del Banco Mundial.*

### Tipo en R y tipo estadístico

R asigna el tipo según lo que encuentra en el archivo, no según lo que la variable
significa. Una clave numérica que representa categorías puede llegar como `<dbl>`.

Sí coinciden  porque las columnas de los años contienen montos económicos y R las asigna correctamente como "dbl", cosa que significa que corresponde perfectamente con una escala numérica continua.

### Lo que `glimpse()` no dice

Lo que noté es que en la tabla y en general, en la lista, las unidades monetarías no esta detalladas, vienen con un lenguaje distinto que hace que se complique un poco la lectura

## 5 · Uso de IA

Ver [Postura sobre IA](https://cjjmdata.github.io/analisis_datos_i/curso/postura-ia.html).

Utilicé Gemini como asistente para indicarme el orden para poner las cosas, tambíen para orientarme y verificar el formato y las dudas con la base de datos del excel del Banco Mundial con R.
También lo use para entender y redactar de forma correcta las respuestas finales de mi archivo.

---

## Cómo entregar

1. Entorno de ejecución → **Reiniciar y ejecutar todo**.
2. Revisa que ninguna celda termine en error y que se vea la salida de `glimpse()`.
   Si subiste el archivo por el panel y la sesión se reinició, súbelo otra vez
   antes de ejecutar.
3. Archivo → Descargar → **Descargar .ipynb**.
4. Sube ese archivo a la actividad en Brightspace.